runtime/v2/shim.go

```diff
@@ -61,23 +65,18 @@ func loadAddress(path string) (string, error) {
 	return string(data), nil
 }
 
-func loadShim(ctx context.Context, bundle *Bundle, onClose func()) (_ ShimInstance, err error) {
+func loadShim(ctx context.Context, bundle *Bundle, onClose func()) (_ ShimInstance, retErr error) {
 	address, err := loadAddress(filepath.Join(bundle.Path, "address"))
 	if err != nil {
 		return nil, err
 	}
-	conn, err := client.Connect(address, client.AnonReconnectDialer)
-	if err != nil {
-		return nil, err
-	}
-	defer func() {
-		if err != nil {
-			conn.Close()
-		}
-	}()
+
+	// TODO: figure out how to negotiate protocol between containerd and shim
+	isGRPC := strings.Contains(strings.ToLower(address), "grpc")
```

Comment:
Including the protocol information in the address string (like URL) may work. 

We could have something like `grpc+unix://` or `grpc+file://` (see https://github.com/whatwg/url/issues/230).

Reply:
Had exactly the same conversation with @dmcgowan 
It'd more useful to return a JSON struct in stdout, so we can explicitly specify which protocol to use.
This will give more room for some new features planned (like shims can now return a list of supported capabilities).
Updated the PR.